llumnix/backends/vllm/llm_engine.py

```diff
@@ -234,14 +255,16 @@ def __init__(
         self._run_workers("init_migration", instance_id=instance_id, migration_config=migration_config,\
                                                       src_worker_handle_list=self.worker_handle_list,
                                                       placement_group=placement_group, node_id=node_id)
-        self._thread = threading.Thread(
-            target=self._start_engine_loop, args=(), daemon=True, name="engine_loop"
+        self.engine_step_loop_thread = threading.Thread(
+            target=self._start_engine_step_loop, args=(), daemon=True, name="engine_step_loop"
         )
-        self._thread.start()
+        self.engine_step_loop_thread.start()
 
-    def _start_engine_loop(self) -> None:
+    def _start_engine_step_loop(self) -> None:
         while True:
-            self.engine.step()
+            num_request_outputs = self.engine.step()
+            if num_request_outputs == 0:
+                time.sleep(0.01)
```

Comment:
Sleep is not a good choice. Consider marking it as a TODO for event-driven.

Reply:
It's a good suggestion. For simplicity, we temporarily use this implementation. We will consider refining the implementation in the future.